packages/protocol/test/team/TimelockTokenPool.t.sol

```diff
@@ -547,4 +547,125 @@ contract TestTimelockTokenPool is TaikoTest {
         assertEq(tko.balanceOf(Alice), 10_000e18);
         assertEq(usdc.balanceOf(Alice), 1_000_000_000e6 - paidUsdc);
     }
+
+    // Let's calculate at which date, how much is unlocked vs. withdrawable.
+    function test_realistic_scenario() public {
```

Comment:
The test doesn't really cover the real world grant settings. Assume we hire Alice at TGE, granting her 1 million TKO.

- grantStart = TGE
- grantCliff: 90 days
- grantPeriod: 4 years
- unlockPeriod: 4 years.

Note that:

- there is no shared unlock start data. For each token (wei), its unlocking starts when this token is vested.
- Alice will own all these tokens at TGE+4 years
- Alice can withdraw all these tokens at TGE+8 years (not TGE+4 years).

Reply:
This is in the token grant letter (or maybe it is in just mine?)
`Tokens that are allocated to the Company at Token Launch will only be available for delivery to you as and when such Tokens are released from the mandatory lockup restriction ("Token Unlock"), which is expected to occur from time to time during the period commencing on the date of Token Launch and ending on the fourth anniversary of the date of Token Launch`

So for Alice (in my example above) , her lockup starts (with some cliff ofc) at the TGE.

In your example, why you say Alice can only withdraw first her tokens after 4 years ? (and 8 years the last portion).

If this is the case, it has very serious implicaitons on token grants.

Reply:
In you case the only thing we need to do is to set the unlock cliff to 1 year, and then that is it.